src/java/org/apache/cassandra/tools/nodetool/stats/TableStatsHolder.java

```diff
@@ -391,6 +391,15 @@ private void initializeKeyspaces(NodeProbe probe, boolean ignore, List<String> t
         }
     }
 
+    private double toDouble(Object valueObj) {
```

Comment:
I am not too big fan of this method. Don't we know already that these metrics are of type double looking where they are produced? Look at "TableMetrics" class for e.g. "CompressionRatio" there. We know this is a double already and we are not supposed to use nodetool of one version with Cassandra node of another. We see that it was cast in StatsTableComparator to Double because we know it was it like that.

For exatra safety, once we cast it to Double, it might be `null`, there are null checks in this class if you see around so I would apply same logic here and remove this method. 

You also need to fix `TableStatsPrinterTest` class.

Reply:
@smiklosovic done, removed `toDouble` and fixed `TableStatsPrinterTest`. Let me know how it looks.

As you noted, it’s likely safe to assume that `CompressionRatio` is of the `double` type because that’s how it’s defined in `TableMetrics.compressionRatio`. `RecentBloomFilterFalseRatio` is of the `double` type as well which is defined in `BloomFilterMetrics.BloomFilterFalseRatio`.

If you get a chance, could you explain:
1. Why did you find the `toDouble` approach inappropriate? I thought it’s more robust and better semantically.
2. How did you find out that `TableStatsPrinterTest` fails? Did you have to run the full test suite? Or did the CI pipeline catch this?

Reply:
1. I did not think it was necessary
2. I just know that test exists ...